DSA-Campaign-Uplift-Estimation/src/main/java/com/google/sps/servlets/CampaignResultsAlgorithmServlet.java

```diff
@@ -101,9 +104,62 @@ public static double getManualCPCFactor(Entity keywordCampaignEntity, Entity DSA
         return Math.min(ratio, 3);
     }
 
-    public static double getLocationsFactor(Entity keywordCampaignEntity, Entity DSACampaignEntity) {
-        // TODO
-        return 1;
+    public static double getLocationsFactor(Entity keywordCampaignEntity, Entity DSACampaignEntity) throws IOException {
+        // read in US Census data
+        BufferedReader file = new BufferedReader(new FileReader("US_Census_Data_State_Populations.txt"));
+
+        // populate the census data into the hash map
+        HashMap<String, Long> statePopulations = new HashMap<String, Long>();
+
+        // ignore the first line containing column headers
+        file.readLine();
+        
+        // read the remaining 56 lines - 4th element contains the location name, 16th element contains the population number
+        for (int i=0; i<56; i++) {
```

Comment:
Generally I would say to make this a constant, as we don't want [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)). 
But in this case, don't we want to have the BufferedReader automatically read the remaining lines without having to give it a number in case the number ever changes?

Reply:
Done